TestModels/Resource/runtimes/rust/src/operation/get_resources/_get_resources_output.rs

```diff
@@ -0,0 +1,66 @@
+// Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[allow(missing_docs)] // documentation missing in model
+#[non_exhaustive]
+#[derive(::std::clone::Clone, ::std::cmp::PartialEq, ::std::fmt::Debug)]
+pub struct GetResourcesOutput {
+    pub(crate) output: crate::types::simple_resource::SimpleResourceImpl,
```

Comment:
So my instincts tell me this should somehow only refer to the `SimpleResource` trait instead. Is there a way to produce some box of a `dyn SimpleResource` instead without unacceptable overhead? I don't know Rust quite well enough to know the exact solution but I strongly suspect it exists.

This will matter more when we get to the `Extendable` test case, where `@extendable` allows for customer implementations of resources, and therefore HAS to be something like what I'm suggesting. It should probably be backwards compatible to add `@extendable` to a resource, but if non-extendable resources are referenced by their concrete type, it wouldn't work to change it to the trait later.

Reply:
The trouble is converting between that and the Dafny type, what with ownership and whatnot.
I'll bang my head against it a little more.

Reply:
Yes, that's the direction I'm going. 
Box didn't work, so I'm using `Rc<RefCell<dyn SimpleResource>>`

Reply:
Great, that's definitely better. I'm not 100% confident there isn't a better choice that doesn't force reference counting, but I'd rather move forward with this choice if it's correct even if it isn't maximally efficient.

I cut https://github.com/smithy-lang/smithy-dafny/issues/449 as a follow up we should do to check whether some of these kinds of choices will work with typical uses.